### PR TITLE
Added system font recognition for xfce-terminal

### PR DESCRIPTION
**Pull request by DarNCelsius**

### CODE REVIEW

**Comment by DarNCelsius** [comment]
It's a little beefy perhaps. I can't but help wish we had an xml  solution to use for XFCE and Openbox files. Would tidy this up quite a bit. XFCE WM Theme is also in this file, which I had plans to do next.
XFConf is an option too:
xfconf

**Comment by LinusDierheimer** [comment]
Funny thing, i just  pushed XFCE theme detection.

**Comment by DarNCelsius** [comment]
I see it. So this could be done a lot simpler.

**Comment by LinusDierheimer** [comment]
Yes i'm just at reviewing your PR

File: src/modules/terminalfont.c [hunk 1]
```diff
@@ -43,6 +43,71 @@ static void printTerminalFontFromConfigFile(FFinstance* instance, const char* co
...
+    }
+    ffStrbufDestroy(&matchText);
+    ffStrbufDestroy(&absolutePath);
+
+    if(line != NULL)
+        free(line);
+
+    fclose(file);
+
+    if(fontName.chars[0] != '\0')
```
Comment (LinusDierheimer) [inline src/modules/terminalfont.c:103]: `fontName.length > 0`

File: src/modules/terminalfont.c [hunk 2]
```diff
@@ -43,6 +43,71 @@ static void printTerminalFontFromConfigFile(FFinstance* instance, const char* co
     printTerminalFont(instance, font);
 }
 
+static void printXCFETerminal(FFinstance* instance)
+{
+    FFstrbuf fontName;
```
Comment (LinusDierheimer) [inline src/modules/terminalfont.c:48]: With the following suggestions in mind, i'm not sure if we need a strbuf. Possible i'm wrong at this one.

Comment (DarNCelsius) [reply src/modules/terminalfont.c:48]: I was just trying to avoid including strings.h altogether.

Comment (LinusDierheimer) [reply src/modules/terminalfont.c:48]: I am too, but if we never need to modify the buffer we need neither strbufs nor strings.h

File: src/modules/terminalfont.c (hunk 1 above)
Comment (DarNCelsius) [reply src/modules/terminalfont.c:103]: I was using length initially - it failed unless I used Recalculate. This approach worked though.

File: src/modules/terminalfont.c [hunk 3]
```diff
@@ -43,6 +43,71 @@ static void printTerminalFontFromConfigFile(FFinstance* instance, const char* co
...
+        printTerminalFontFromConfigFile(instance, "xfce4/terminal/terminalrc", "FontName=%[^\n]");
+        ffStrbufDestroy(&fontName);
+        return;
+    }
+
+    FFstrbuf absolutePath;
+    ffStrbufInitA(&absolutePath, 64);
+    ffStrbufAppendS(&absolutePath, instance->state.passwd->pw_dir);
+    ffStrbufAppendC(&absolutePath, '/');
+    ffStrbufAppendS(&absolutePath, ".config/xfce4/xfconf/xfce-perchannel-xml/xsettings.xml");
```
[the remark this answers is not in the document]
Comment (DarNCelsius) [reply src/modules/terminalfont.c:64]: Doh

File: src/modules/terminalfont.c [hunk 4]
```diff
@@ -43,6 +43,71 @@ static void printTerminalFontFromConfigFile(FFinstance* instance, const char* co
...
+static void printXCFETerminal(FFinstance* instance)
+{
+    FFstrbuf fontName;
+    ffStrbufInitA(&fontName, 256);
+
+    ffParsePropFileConfig(instance, "xfce4/terminal/terminalrc", "FontUseSystem=%[^\n]", fontName.chars);
+
+    if((fontName.chars[0] == '\0') || !ffStrbufCompS(&fontName, "FALSE"))
+    {
+        printTerminalFontFromConfigFile(instance, "xfce4/terminal/terminalrc", "FontName=%[^\n]");
```
[the remark this answers is not in the document]
Comment (DarNCelsius) [reply src/modules/terminalfont.c:55]: I don't have a fontName at this point though...

File: src/modules/terminalfont.c [hunk 5]
```diff
@@ -43,6 +43,71 @@ static void printTerminalFontFromConfigFile(FFinstance* instance, const char* co
...
+        }
+        ffStrbufClear(&fontName);
+    }
+    ffStrbufDestroy(&matchText);
+    ffStrbufDestroy(&absolutePath);
+
+    if(line != NULL)
+        free(line);
+
+    fclose(file);
```
[the remark this answers is not in the document]
Comment (DarNCelsius) [reply src/modules/terminalfont.c:101]: `ffParsePropFileConfig` fails if the line has different spacing though. That's why I trimmed the result before the compare. It's very fragile if someone decides to edit the file manually.

File: src/modules/terminalfont.c (hunk 2 above)
Comment (DarNCelsius) [reply src/modules/terminalfont.c:48]: I see. Will try. :)

File: src/modules/terminalfont.c (hunk 5 above)
[the remark this answers is not in the document]
Comment (DarNCelsius) [reply src/modules/terminalfont.c:101]: Could the function be modified to ignore leading/trailing whitespace somehow?

[the remark this answers is not in the document]
Comment (DarNCelsius) [reply src/modules/terminalfont.c:101]: Leading would suffice.

Is there a use case for keeping leading white space intact?

File: src/modules/terminalfont.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (LinusDierheimer) [reply src/modules/terminalfont.c:55]: Yea my fault sorry

File: src/modules/terminalfont.c (hunk 5 above)
[the remark this answers is not in the document]
Comment (LinusDierheimer) [reply src/modules/terminalfont.c:101]: I think its worth implementing a ffSettingsGetXFConf at this point using libxfconf. I will look into this soon.

[the remark this answers is not in the document]
Comment (LinusDierheimer) [reply src/modules/terminalfont.c:101]: But as a note, a single white space in the scanf format string counts as any number of white spaces in the scanned string, thats why i prepended a single white space before the actual format string.

[the remark this answers is not in the document]
Comment (LinusDierheimer) [reply src/modules/terminalfont.c:101]: You have waited long enough, here it is!

[the remark this answers is not in the document]
Comment (DarNCelsius) [reply src/modules/terminalfont.c:101]: Nice. The new system looks a lot better too. :+1:

**Comment by DarNCelsius** [comment]
Quite happy with this one, apart from the `string.h` requirement. I couldn't think of an elegant solution.

**Comment by LinusDierheimer** [comment]
This seems fine. There is no problem with including string.h, just with using some dangerous functions out of it. Good resource for this is the commit history of https://github.com/git/git/blob/master/banned.h. I don't see a problem with strcasecmp.

**Pull request merged**